### PR TITLE
added note about field compare

### PR DESCRIPTION
**Pull request by jalogisch**
after realisation that it is possible to compare two fields I add the note about.

need to be merged to 3.0 too

### CODE REVIEW

File: pages/pipelines/rules.rst [hunk 1]
```diff
@@ -96,6 +96,7 @@ The condition must not be empty, but can simply consist of the boolean literal `
 
 If a condition calls a function which is not present (perhaps due to a typo or missing plugin) the call evaluates to ``false``.
 
+.. note:: Comparing two fields can be done when you use the same data type, e.g.  ``to_string($message.src_ip) == to_string($message.dst_ip)`` will compare the two strings and will become ``true`` on match.
```
Comment (mpfz0r) [inline pages/pipelines/rules.rst]: n00b reviewer question: what happens if you try to compare different data types?

Comment (jalogisch) [reply pages/pipelines/rules.rst]: I just tried that and it just fails - because it can't compared.

Comment (mpfz0r) [reply pages/pipelines/rules.rst]: is that visible to the user?

Comment (jalogisch) [reply pages/pipelines/rules.rst]: the `when` condition did not become true - so `then` is not executed. 

Not directly visible, but the rule just did not work.

Comment (mpfz0r) [reply pages/pipelines/rules.rst]: I was wondering if that's noteworthy.  e.g  "Comparing different data types evaluates to ``false``".

Comment (jalogisch) [reply pages/pipelines/rules.rst]: yes, good catch!

**Review by mpfz0r — APPROVED** [review]

**Pull request merged**